python/lsst/daf/butler/formatters/jsonFormatter.py

```diff
@@ -78,7 +78,50 @@ def _writeFile(self, inMemoryDataset, fileDescriptor):
         with open(fileDescriptor.location.path, "w") as fd:
             if hasattr(inMemoryDataset, "_asdict"):
                 inMemoryDataset = inMemoryDataset._asdict()
-            json.dump(inMemoryDataset, fd)
+            fd.write(self._toBytes(inMemoryDataset).decode())
+
+    def _fromBytes(self, bytesObject, pytype=None):
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        serializedDataset : `bytes`
```

Comment:
It's called `bytesObject` in the parameter list but I think I prefer `serializedDataset` so use that for the parameter list.